go-controller/pkg/ovn/obj_retry.go

```diff
@@ -0,0 +1,128 @@
+package ovn
+
+import (
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type retryObjEntry struct {
+	newObj     interface{}
+	oldObj     interface{}
+	config     interface{}
+	timeStamp  time.Time
+	backoffSec time.Duration
+	// whether to include this object in the retry iterations
+	ignore bool
+}
+
+// initRetryObj tracks an object that failed to be created to potentially retry later
+// initially it is marked as skipped for retry loop (ignore = true)
+func (r *retryObjs) initRetryObj(obj interface{}, key string) {
+	r.retryMutex.Lock()
+	defer r.retryMutex.Unlock()
+	if entry, ok := r.entries[key]; ok {
+		entry.timeStamp = time.Now()
+		entry.newObj = obj
+	} else {
+		r.entries[key] = &retryObjEntry{newObj: obj, oldObj: nil, config: nil,
+			timeStamp: time.Now(), backoffSec: 1, ignore: true}
+	}
+}
+
+// initRetryWithDelete tracks an object that failed to be deleted to potentially retry later
+// initially it is marked as skipped for retry loop (ignore = true)
+func (r *retryObjs) initRetryObjWithDelete(obj interface{}, key string, config interface{}) {
```

Comment:
Can we combine these 2 functions initRetryObjWithDelete + initRetryObj ? 
Or maybe rename initRetryObj to initRetryObjWithAdd ?
Feels like they do almost exaclt the same think besides the `oldObj` vs `newObj` part.

Reply:
that is exactly right one handle add path and another for delete I can rename with Add prefix